packages/shared/src/browser.ts

```diff
@@ -6,6 +11,27 @@ export function inBrowser(): boolean {
   return typeof window !== 'undefined';
 }
 
+const rawBrowserGlobals: BrowserGlobals = {
+  document: window?.document,
+  window: window,
+};
+
+/**
+ *
+ * Returns the browser globals if the current environment is a browser. Else, it will throw an error upon access.
+ * @returns {BrowserGlobals}
+ * @throws {Error} - If the current environment is not a browser.
+ */
+export const browser = inBrowser()
+  ? rawBrowserGlobals
+  : new Proxy(rawBrowserGlobals, {
```

Comment:
@tmilewski Could you please share some context here? Have we set up any specific test environments / eslint config to catch issues like these without testing with an actual expo app?

Unless I'm missing something, trying to access these in non-browser environments would result in errors even without the Proxy here.